pkg/common/utils/utils.go

```diff
@@ -61,3 +61,37 @@ func QueryVolumeUtil(ctx context.Context, m cnsvolume.Manager, queryFilter cnsty
 	}
 	return queryResult, nil
 }
+
+// QueryAllVolumeUtil helps to invoke query volume API based on the feature state set for using query async volume.
+// If useQueryVolumeAsync is set to true, the function invokes CNS QueryVolumeAsync, otherwise it invokes synchronous QueryAllVolume API
+// The function also take volume manager instance, query filters, query selection as params
+// Returns queryResult when query volume succeeds, otherwise returns appropriate errors
+func QueryAllVolumeUtil(ctx context.Context, m cnsvolume.Manager, queryFilter cnstypes.CnsQueryFilter, querySelection cnstypes.CnsQuerySelection, useQueryVolumeAsync bool) (*cnstypes.CnsQueryResult, error) {
```

Comment:
Both these util methods have a lot of common code. Will it be better if we pass another `bool` from the caller and invoke `QueryVolume` or `QueryAllVolume` based on the flag?

Reply:
It adds complexity with the current boolean var `useQueryVolumeAsync` and we may have to handle 4 conditions in a single method to make it robust. Thought I will keep it this way for simplicity

Reply:
Why don't we read `useQueryVolumeAsync` FSS in the util method itself, and not pass it from the caller? We should be able to do it for both syncer and driver, and that should simplify things.

Reply:
this new utils is inside `common` package and we don't want to introduce fss initialization in these packages

Reply:
Or if this change is needed fast, we can take up the refactoring in subsequent PR. Approving this, but can you track the refactoring task internally or as a github issue?